src/App.tsx

```diff
@@ -15,71 +17,65 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+export const App: React.FC = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
+  const messege = selectedGood.length > 0 ? `${selectedGood} is selected`
+    : 'No goods selected';
+  // const buttonDel = document.getElementsByClassName('delete ml-3');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  return (
+    <main className="section container">
+      <h1 className={cn('title', {
+        'title is-flex is-align-items-center': selectedGood.length > 0,
+      })}
+      >
+        {messege}
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+        {selectedGood.length > 0 && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => {
+              setSelectedGood('');
+            }}
+          />
+        )}
+      </h1>
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              data-cy="Good"
+              className={cn('', {
+                'has-background-success-light': selectedGood === good,
+              })}
             >
-              +
-            </button>
-          </td>
+              <td>
+                <button
+                  data-cy={`${selectedGood === good ? 'RemoveButton' : 'AddButton'}`}
+                  type="button"
+                  className={cn('button', {
+                    ' is-info': selectedGood === good,
+                  })}
+                  onClick={
+                    selectedGood === good
+                      ? () => setSelectedGood('')
+                      : () => setSelectedGood(good)
+                  }
+                >
+                  {selectedGood === good ? '-' : '+'}
```

Comment:
found 4 places where you use 
selectedGood === good
consider to make variable for example `const isSelected = selectedGood === good`
and use in all places